tests/tests/test_mgmt_tag_attributes.py

```diff
@@ -0,0 +1,165 @@
+        management_client.updateTagAttributes(did, tags_body)
+
+        res = management_client.getDevice(did)
+        etag_one = res['tags_etag']
+        management_client.updateTagAttributes(did, tags_body, eTag=etag_one)
+
+        res = management_client.getDevice(did)
+        etag_two = res['tags_etag']
+        assert etag_one != etag_two
+
+        tags_attributes = []
+        for attr in res['attributes']:
+            if attr['scope'] == 'tags':
+                assert attr['name'] in tags
+                tags_attributes.append(attr)
+        assert len(tags_attributes) == len(tags_body)
+
+    def test_replace_tag_attributes_with_etag(
+            self,
+            management_client,
+            internal_client,
+            inventory_attributes):
+        did = "some-device-id"
+        tags = {"n_4": {"name": "n_4", "value": "v_4", "description": "desc_4"}}
+        tags_body = [tags['n_4']]
+        internal_client.create_device(did, inventory_attributes)
+        management_client.setTagAttributes(did, tags_body)
+
+        res = management_client.getDevice(did)
+        etag_one = res['tags_etag']
+        management_client.setTagAttributes(did, tags_body, eTag=etag_one)
+
+        res = management_client.getDevice(did)
+        etag_two = res['tags_etag']
+        assert etag_one != etag_two
+
+        tags_attributes = []
+        for attr in res['attributes']:
+            if attr['scope'] == 'tags':
+                assert attr['name'] in tags
+                tags_attributes.append(attr)
+        assert len(tags_attributes) == len(tags_body)
+
+    def test_update_tag_attributes_with_wrong_etag(
+            self,
+            management_client,
+            internal_client,
+            inventory_attributes):
+        did = "some-device-id"
+        tags = {"n_5": {"name": "n_5", "value": "v_5", "description": "desc_5"}}
+        tags_body = [tags['n_5']]
+        internal_client.create_device(did, inventory_attributes)
+        management_client.updateTagAttributes(did, tags_body)
+
+        fake_etag = "241496e0-cbbb-4a83-90e9-70b4dd0e645a"
+        try:
+            management_client.updateTagAttributes(did, tags_body, eTag=fake_etag)
+        except Exception as e:
+            assert str(e) == "412 Precondition Failed"
+
+    def test_replace_tag_attributes_with_wrong_etag(
+            self,
+            management_client,
+            internal_client,
+            inventory_attributes):
+        did = "some-device-id"
+        tags = {"n_6": {"name": "n_6", "value": "v_6", "description": "desc_6"}}
+        tags_body = [tags['n_6']]
+        internal_client.create_device(did, inventory_attributes)
+        management_client.setTagAttributes(did, tags_body)
+
+        fake_etag = "241496e0-cbbb-4a83-90e9-70b4dd0e645a"
+        try:
+            management_client.setTagAttributes(did, tags_body, eTag=fake_etag)
+        except Exception as e:
+            assert str(e) == "412 Precondition Failed"
```

Comment:
```suggestion
        try:
            management_client.setTagAttributes(did, tags_body, eTag=fake_etag)
        except Exception as e:
            assert str(e) == "412 Precondition Failed"
        else:
            raise Exception("did not raise expected exception")
```

Reply:
@tranchitella I thought it is enough to fail on assert in this case. It looks something like:
```
tests/test_mgmt_tag_attributes.py:165: in test_replace_tag_attributes_with_wrong_etag
     assert str(e) == "412 Precondition Failed"
 E   AssertionError: assert '500 Internal Server Error' == '412 Precondition Failed'
 E     - 412 Precondition Failed
 E     + 500 Internal Server Error
```

Reply:
Ah, it's for the case when no any exception raised. Makes sense. Thanks!